misc/python/materialize/mzcompose/composition.py

```diff
@@ -753,6 +754,17 @@ def sanity_restart_mz(self) -> None:
                 print("Service materialized not running, will not restart it.")
                 return
 
+            for (db,) in self.sql_query("SELECT name FROM mz_databases"):
+                self.sql(f"DROP DATABASE {identifier(db)}", user="mz_system", port=6877)
```

Comment:
I think this is a bit insidious -- in a sanity check we perform destructive DDL. People are very likely to get bitten by this if their test actually attempts to execute multiple workflows in sequence.

If an unrelated database , e.g. `sanity_check` was being created and dropped as part of the sanity check, would that  have exposed the same issue?